plugins/modules/system/sysrc.py

```diff
@@ -0,0 +1,277 @@
+            - Use I(absent) to remove the variable.
+            - Use I(value_present) to add the value to the existing variable.
+            - Use I(value_absent) to remove the value from the existing variable.
+        type: str
+        default: "present"
+        choices: [ absent, present, value_present, value_absent ]
+    path:
+        description:
+            - Path to file to use instead of C(/etc/rc.conf).
+        type: str
+        default: "/etc/rc.conf"
+    delim:
+        description:
+            - Delimiter to be used insetad of C( )
+            - Only used when I(state=value_present) or I(state=value_absent).
+        default: " "
+        type: str
+    jail:
+        description:
+            - Name or ID of the jail to operate on
+        required: false
+        type: str
+notes:
+  - The C(name) cannot use . (periods) as sysrc doesn't support OID style names
+'''
+
+EXAMPLES = r'''
+---
+# enable mysql in the /etc/rc.conf
+- name: Configure mysql pid file
+  sysrc:
+    name: mysql_pidfile
+    value: "/var/run/mysqld/mysqld.pid"
+
+# enable accf_http kld in the boot loader
+- name: enable accf_http kld
+  sysrc:
+    name: accf_http_load
+    state: present
+    value: "YES"
+    path: /boot/loader.conf
+
+# add gif0 to cloned_interfaces
+- name: add gif0 interface
+  sysrc:
+    name: cloned_interfaces
+    state: value_present
+    value: "gif0"
+
+# enable nginx on a jail
+- name: add gif0 interface
+  sysrc:
+    name: nginx_enable
+    value: "YES"
+    jail: www
+'''
+
+RETURN = r'''
+changed:
+  description: Return changed for sysrc actions as true or false.
+  returned: always
+  type: bool
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+import re
+
+
+class sysrc(object):
+    def __init__(self, module, name, value, path, delim, jail):
+        self.module = module
+        self.name = name
+        self.changed = False
+        self.value = value
+        self.path = path
+        self.delim = delim
+        self.jail = jail
+        self.sysrc = module.get_bin_path('sysrc', True)
+
+    def has_unknown_variable(self, out, err):
+        # newer versions of sysrc use stderr instead of stdout
+        return err.find("unknown variable") > 0 or out.find("unknown variable") > 0
+
+    def exists(self):
+        # sysrc doesn't really use exit codes
+        (rc, out, err) = self.run_sysrc(self.name)
+        if self.value is None:
+            regex = "%s: " % re.escape(self.name)
+        else:
+            regex = "%s: %s$" % (re.escape(self.name), re.escape(self.value))
+
+        return not self.has_unknown_variable(out, err) and re.match(regex, out) is not None
+
+    def contains(self):
+        (rc, out, err) = self.run_sysrc('-n', self.name)
+        if self.has_unknown_variable(out, err):
+            return False
+
+        return self.value in out.strip().split(self.delim)
+
+    def create(self):
+        if self.module.check_mode:
+            self.changed = True
+            return
```

Comment:
Wouldn't it be consistent to return here as well `True` like in the regular case? Right now this will return `None`. I know you do not use the return value anywhere but in case you would it would not result in different behaviour.

Reply:
I also updated the other functions that were doing the same thing